instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -729,7 +725,20 @@ async def __call__(
                 )
 
                 await self.app(scope, otel_receive, otel_send)
+        except Exception as exc:  # pylint: disable=broad-except
+            exception = exc
         finally:
+            if exception is not None and _report_new(
```

Comment:
Just a style thing, but I'm wondering if it would be worthwhile to break this method into functions for readability. Perhaps those functions could stand to be further modularized, but this would be a step in that direction. Something like
```
        try:
            await self.instrument_app(attributes, receive, scope, send, span, span_name)
        except Exception as exc:  # pylint: disable=broad-except
            exception = exc
        finally:
            await self.cleanup(active_requests_count_attrs, attributes, exception, scope, span, start, token)
```
Not a blocking comment, just a suggestion while we're in here.

Reply:
Another suggestion would also to wait and implement this for the various instrumentations that still require this (like flask, wsgi (for metrics), django, etc) and see if there are any commonalities we can refactor out. Difficult to say what we can abstract/modularize without seeing more implementation.

Reply:
Agreed. Indeed, while reviewing the transition PRs I identified several instrumentations we are not handling exceptions and setting the exception in `error.type` even for the ones that are already migrated to the new semconv. I will mark this as draft for now while I think in a better way to handle this. Maybe we'll have a better idea on the commonalities after more transitions.